tests/Oryx.BuildImage.Tests/Hugo/HugoDynamicInstallationTest.cs

```diff
@@ -0,0 +1,125 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Oryx.BuildScriptGenerator.Hugo;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Oryx.BuildImage.Tests
+{
+    public class HugoDynamicInstallationTest : HugoSampleAppsTestBase
+    {
+        private readonly string DefaultInstallationRootDir = HugoConstants.InstalledHugoVersionsDir;
+
+        public HugoDynamicInstallationTest(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        public static TheoryData<string> ImageNameData
+        {
+            get
+            {
+                var imageTestHelper = new ImageTestHelper();
+                var data = new TheoryData<string>();
+                data.Add(imageTestHelper.GetAzureFunctionsJamStackBuildImage());
+                data.Add(imageTestHelper.GetGitHubActionsBuildImage());
+                return data;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ImageNameData))]
+        public void GeneratesScript_AndBuilds(string imageName)
+        {
+            // Arrange
+            var version = "0.59.1";
+            var installationDir = $"{BuildScriptGenerator.Constants.TemporaryInstallationDirectoryRoot}/hugo/{version}";
+            var appName = SampleAppName;
+            var volume = CreateSampleAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = "/tmp/app-output";
+            var script = new ShellScriptBuilder()
+                .SetEnvironmentVariable("HUGO_VERSION", version)
```

Comment:
Users can set this environment variable in their github action to dynamically install the version they want

Reply:
so we default to 0.59.1, but user can override it?